### PR TITLE
 Semantic versioning for osmosis v5.x

### PR DESCRIPTION
**Pull request by mattverse**
Closes: #[621](https://github.com/osmosis-labs/osmosis/issues/621)

## Description

This PR adds semantic versioning for v5 in the imports. It also includes change in `generate-proto.sh` to ensure proto files get generated in the proper directory.

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: scripts/protocgen.sh [hunk 1]
```diff
@@ -29,5 +29,5 @@ Mgoogle/protobuf/any.proto=github.com/cosmos/cosmos-sdk/codec/types:. \
   $(find "${dir}" -maxdepth 1 -name '*.proto')
 done
 
-cp -r ./github.com/osmosis-labs/osmosis/* ./
+cp -r ./github.com/osmosis-labs/osmosis/v*/x/* x/
```
Comment (ValarDragon) [inline scripts/protocgen.sh:32]: Oh cool, did you test make proto-gen on this branch and it works?

**Review by ValarDragon — APPROVED** [review]
LGTM! Thanks for doing this. Lets merge once `make proto-gen` is confirmed to work / not work in the new setup

File: scripts/protocgen.sh (hunk 1 above)
Comment (mattverse) [reply scripts/protocgen.sh:32]: Yep! With this included, proto doesn't get confused where it should generate the proto files even with the go semantic versioning included!

Comment (faddat) [reply scripts/protocgen.sh:32]: yup

**Review by mattverse — COMMENTED** [review]
I think just a few changes should land before merge!
- ensure cosmos-sdk stays at v0.42.9 at go.mod
- delete `github.com/osmosis-labs/osmosis/v4` directory that has been newly made, these seem to have been created when generated proto without setting paths in the proto gen script file.

**Pull request merged**